### DISCUSSION TITLE
Call recording - external storage providers

### DISCUSSION THREAD

**Discussion by tbarbugli** [GetStream/protocol#371] [Q&A]
At the moment Stream Video allows you to store recorded calls on S3, Google Cloud Storage and Azure Blob Storage. You can use this conversation to suggest storage options that are important for your application.

**Updates**

- 13 May 2015: Added support for S3 compatible storage (eg. [minIO](https://min.io/))

**Comment by mactunechy** [comment]
the external storage api is throwing an error: 500. here is my code example:

```js

import { StreamClient } from "@stream-io/node-sdk";

const client = new StreamClient(streamApiKey, streamSecret);

const result = await client.video.createExternalStorage({
  bucket: "xxxx",
  name: "my-s3",
  storage_type: "s3",
  path: "/",
  aws_s3: {
    s3_region: "eu-north-1",
    s3_api_key: "xxx",
    s3_secret: "xxx",
  },
});
```

**Reply by jbrainz** [reply]
It would be really nice if you can provide support for cloudinary too.

**Reply by Aanu1995** [reply]
External storage not working. it was successful created but the recorded video was still stored in getStream storage

**Reply by flex-akin** [reply]
hii @Aanu1995 , can you give insight on how to retrieve the recordings from the getstream storage, because i keep getting none.
i am using flutter

**Reply by Aanu1995** [reply]
@flex-akin You can follow the documentation on how to retrieve the recordings here https://getstream.io/video/docs/api/recording/calls/. I don't use this API because our own goal is to have the link to the recording in storage stored in our database. So, I am using web hook to know when recording is ready and retrieve the data to be stored in our database.

**Reply by Aanu1995** [reply]
@tbarbugli Can you help on how to make the recording use my S3 external storage. I already followed the documentation and I was able to add the storage successfully but recordings are still being stored in GetStream storage instead of the external storage I provided. Thank you

**Reply by drexhacker** [reply]
@Aanu1995 Me I opted for using the webhook so when the recording is ready I pull it to my own storage. I think this a very easy temporary workaround, which gets the job done

**Reply by Aanu1995** [reply]
Thank you. That was exactly what I did too

**Reply by tbarbugli** [reply]
this was resolved a while ago, please ping here if you still have issues with external storage

**Reply by Sahil1849** [reply]
> this was resolved a while ago, please ping here if you still have issues with external storage

yes, i am still facing that issue

**Comment by viscafcb96** [comment]
Can i use Digital Ocean Space or Cloudflare R2 as a storage?

**Reply by trmcnvn** [reply]
As these are both S3-API compatible, could the AWS S3 storage provider be expanded to customize the endpoint?

**Reply by joostvanhoof** [reply]
@tbarbugli I had the same question as @Phu96, I would like to use DigitalOcean Spaces as an external storage provider. As @trmcnvn indicated these are S3-API compatible. Any timeline you can share?

**Reply by tbarbugli** [reply]
thank you for the input, we are going to discuss adding this in Q1

**Reply by caspar-goeke** [reply]
@tbarbugli we also would like to store it on R2, which is s3 compatible (but much cheaper). Any updates on this?

**Reply by joostvanhoof** [reply]
@tbarbugli the docs do not list any other cloud providers if I'm correct. Any update on the timeline for this? I'm looking for DigitalOcean Spaces support (S3 storage).

**Reply by Sankarraoheits** [reply]
i also would like to store in R2, any update on this?

**Comment by oddanderson** [comment]
+1 for Supabase

**Comment by teetangh** [comment]
how does one store the recordings in supabase ?

**Comment by sardor01** [comment]
Any plans to support cloudflare?

**Reply by zibrohimov** [reply]
yes please
